ui/components/component-library/popover/popover.stories.tsx

```diff
@@ -769,6 +769,49 @@ export const OnPressEscKey: StoryFn<typeof Popover> = (args) => {
   );
 };
 
+export const OnClickOutside: StoryFn<typeof Popover> = (args) => {
+  const [referenceElement, setReferenceElement] = useState();
+  const [isOpen, setIsOpen] = useState(false);
+
+  // Set Popover Ref
+  const setBoxRef = (ref) => {
+    setReferenceElement(ref);
+  };
+
+  const handleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const handleClickOutside = () => {
+    setIsOpen(false);
+  };
+
+  return (
+    <>
+      <Box
+        ref={setBoxRef}
+        onClick={handleClick}
+        backgroundColor={BackgroundColor.primaryAlternative}
+        style={{ width: 200, height: 200 }}
+        color={TextColor.primaryInverse}
+        as="button"
+      >
+        Click to open
+      </Box>
+      <Popover
+        referenceElement={referenceElement}
+        onClickOutside={handleClickOutside}
+        isOpen={isOpen}
+        {...args}
+      >
+        Click outside to close
+      </Popover>
+    </>
+  );
+};
+
+OnClickOutside.storyName = 'onClickOutside';
+
 export const WithPopoverHeader: StoryFn<typeof Popover> = (args) => {
   const [refTitleElement, setRefTitleElement] = useState();
   const [isOpen, setIsOpen] = useState(true);
```

Comment:
Non blocking: Can we remove the console log and it looks like there is a type error for color in the `WithPopoverHeader` story

<img width="1470" alt="Screenshot 2023-11-09 at 7 45 37 AM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/884c8255-52b9-4255-b678-30b4adaa3c4f">